crates/nu-command/src/viewers/table.rs

```diff
@@ -1552,13 +1551,9 @@ impl Iterator for PagingTableCreator {
             batch.push(item);
             idx += 1;
 
-            if idx % STREAM_TIMEOUT_CHECK_INTERVAL == 0 {
-                let end_time = Instant::now();
-
-                // If we've been buffering over a second, go ahead and send out what we have so far
-                if (end_time - start_time).as_secs() >= 1 {
-                    break;
-                }
+            // If we've been buffering over a second, go ahead and send out what we have so far
+            if (Instant::now() - start_time).as_secs() >= 1 {
```

Comment:
Start time needs to be reset at some point, else it will always break after exhausting the first second.

Reply:
Start time is reset for every page of records (because this is happening in `PagingTableCreator::next()`).

Reply:
Right, read that blindly as just the inner loop.